packages/expo-notifications/src/Notifications.types.ts

```diff
@@ -268,77 +314,42 @@ export type CalendarTriggerInputValue = {
- * @platform ios
+ * See [`SchedulableNotificationTriggerInput`](#schedulablenotificationtriggerinput)
  */
-export type CalendarTriggerInput = CalendarTriggerInputValue & {
+export type SchedulableNotificationTriggerInputObject = {
+  type: CalendarTriggerTypes;
   channelId?: string;
   repeats?: boolean;
+  seconds?: number;
+  timezone?: string;
+  year?: number;
+  month?: number;
+  weekday?: number;
+  weekOfMonth?: number;
+  weekOfYear?: number;
+  weekdayOrdinal?: number;
+  day?: number;
+  hour?: number;
+  minute?: number;
+  second?: number;
 };
 
 /**
- * A trigger that will cause the notification to be delivered once or many times (depends on the `repeats` field) after `seconds` time elapse.
- * > **On iOS**, when `repeats` is `true`, the time interval must be 60 seconds or greater. Otherwise, the notification won't be triggered.
- */
-export interface TimeIntervalTriggerInput {
-  channelId?: string;
-  repeats?: boolean;
-  seconds: number;
-}
-
-/**
- * A trigger that will cause the notification to be delivered once per day.
- */
-export interface DailyTriggerInput {
-  channelId?: string;
-  hour: number;
-  minute: number;
-  repeats: true;
-}
-
-/**
- * A trigger that will cause the notification to be delivered once every week.
- * > **Note:** Weekdays are specified with a number from `1` through `7`, with `1` indicating Sunday.
- */
-export interface WeeklyTriggerInput {
-  channelId?: string;
-  weekday: number;
-  hour: number;
-  minute: number;
-  repeats: true;
-}
-
-/**
- * A trigger that will cause the notification to be delivered once every year.
- * > **Note:** all properties are specified in JavaScript Date's ranges.
- */
-export interface YearlyTriggerInput {
-  channelId?: string;
-  day: number;
-  month: number;
-  hour: number;
-  minute: number;
-  repeats: true;
-}
-
-/**
- * A trigger that will cause the notification to be delivered once at the specified `Date`.
- * If you pass in a `number` it will be interpreted as a Unix timestamp.
- */
-export type DateTriggerInput = Date | number | { channelId?: string; date: Date | number };
-
-/**
- * A type represents time-based, schedulable triggers. For these triggers you can check the next trigger date
+ * Input for time-based, schedulable triggers.
+ * For these triggers you can check the next trigger date
  * with [`getNextTriggerDateAsync`](#notificationsgetnexttriggerdateasynctrigger).
+ * If you pass in a `number` (Unix timestamp) or `Date`, it will be processed as a
+ * trigger input of type [`CalendarTriggerTypes.DATE`](#date). Otherwise, the input must be
+ * an object, with a `type` value set to one of the allowed values in
+ * [`CalendarTriggerTypes`](#calendartriggertypes).
+ * If the input is an object, date components passed in will be validated, and
+ * an error thrown if they are outside their allowed range (e.g. the `minute` and
```

Comment:
```suggestion
 * an error is thrown if they are outside their allowed range (for example, the `minute` and
```

Let's avoid using Latin abbreviations as per our writing styleguide.